batavia/types/Str.js

```diff
@@ -198,7 +198,19 @@ Str.prototype.__ge__ = function(other) {
 }
 
 Str.prototype.__contains__ = function(other) {
-    return false
+    var types = require('../types')
+    if (types.isinstance(other, [
+        types.Bool, types.Int, types.Float,
+        types.List, types.Dict, types.Tuple,
+        types.Set, types.Bytearray, types.Bytes,
+        types.Type, types.Complex, types.NotImplementedType,
+        types.Range, types.Slice, types.FrozenSet
+
+    ])) {
+        throw new exceptions.TypeError.$pyclass('cannot search type ' + type_name(other) + ' in string')
```

Comment:
Yes - the error message is probably incorrect; the test proposed here follows an example from earlier in the suite for startswith. `in` should probably be added to the list of operators to be tested.

And yes, the condition could probably be simplified too. Fancy making a pull request? :-)